Code-4-Community/speak-for-the-trees-frontend
# Allows users to name trees they adopt

## Why

[ClickUp Ticket](https://app.clickup.com/t/2b937c2)


Allow users to name trees (specifically, site entries) that they have adopted (and admin users any trees).

## This PR


- Add name site entry route to available routes
- Combine the desktop and mobile styled components in `pagerHeader` into one and move them to `themedComponents`
- Create the `editTreeNameForm` that allows users to (re)name trees, if allowed:
    - Admin users can (re)name any tree
    - Non-admin users can (re)name a tree only if they have adopted the tree
- Create the `treePageHeader` component that displays the tree's name (along with its common name and the site's address)

## Screenshots

 
If the user is an admin or an adopter and the tree has no name:
![admin-or-adopter-no-name](https://user-images.githubusercontent.com/33704204/208352873-bd9dcce2-a52b-4e7a-a6b6-dd9cd04b8bcd.png)

If the user is not an admin or an adopter and the tree has no name:
![non-adopter-no-name](https://user-images.githubusercontent.com/33704204/208352881-6d7cf89e-a89e-4648-840c-3bb9b245950a.png)

If the user is an admin or adopter and the tree has a name:
![admin-name](https://user-images.githubusercontent.com/33704204/208353024-445a043a-bd92-4d03-aaba-8fd1db45421a.png)

(when the admin or adopter clicks on the pencil icon):
![admin-name-change](https://user-images.githubusercontent.com/33704204/208353224-98a60d04-24b8-4c23-9e10-0508dd0afc17.png)

If the user is not an admin or an adopter and the tree has a name:
![non-adopter-name](https://user-images.githubusercontent.com/33704204/208353069-6c82acdb-352d-420a-95cc-5d6fe222e9e5.png)

A few screenshots on mobile:
![mobile-1](https://user-images.githubusercontent.com/33704204/208353241-29472efa-73a7-48d1-92be-2ea2cb5cb0f1.png)

![mobile-2](https://user-images.githubusercontent.com/33704204/208353244-532c2734-3d88-4e10-8728-279ef2a326f0.png)

## Verification Steps

 
Visually verified to make sure that the texts/icons/buttons appear as expected in the above cases (in the screenshot).
Changed names of trees to several different values (including empty string to delete its name) and checked the database to make sure that the changes were reflected there as appropriate.
